cdap-app-fabric/src/main/java/co/cask/cdap/app/preview/PreviewStatus.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.app.preview;
+
+import java.util.Objects;
+import javax.annotation.Nullable;
+
+/**
+ * Class representing status and failure reason (if any) associated with the Preview.
+ */
+public class PreviewStatus {
+  /**
+   * Status for the preview
+   */
+  public enum Status {
+    RUNNING,
+    COMPLETED,
+    DEPLOY_FAILED,
+    RUN_FAILED
+  }
+
+  private final Status status;
+  private final String failureMessage;
```

Comment:
why the failure is just a string message instead of `Throwable` (which contains the stacktrace)?